block-streamer/src/control.rs

```diff
@@ -33,17 +37,23 @@ pub async fn control(listen_addr: &String) -> Result<()> {
             }
         }
 
-        if let Ok(msg) = from_slice::<ApplicationAPI>(&buf[..real_len]) {
-            match msg {
-                ApplicationAPI::Receive { path, listen_addr } => {
-                    receive(&PathBuf::from(path), &listen_addr).await?
-                }
-                ApplicationAPI::Transmit { path, target_addr } => {
-                    transmit(&PathBuf::from(path), &target_addr).await?
-                }
-                other => {
-                    info!("Received {:?}", other);
-                }
+        let mut databuf = &buf[..real_len];
+        match Message::decode(&mut databuf) {
+            Ok(Message::ApplicationAPI(ApplicationAPI::Receive { listen_addr })) => {
+                receive(&listen_addr).await?
+            }
+            Ok(Message::ApplicationAPI(ApplicationAPI::Transmit { path, target_addr })) => {
+                transmit(&PathBuf::from(path), &target_addr).await?
+            }
+            Ok(Message::DataProtocol(data_msg)) => {
+                data_receiver.handle_transmission_msg(data_msg).await?;
+                data_receiver.attempt_tree_assembly().await?;
+            }
+            Ok(message) => {
+                info!("Received unhandled message: {:?}", message);
```

Comment:
Is this an error case as well (vs info)?

Reply:
This may become an error case later, but today it is more of debug info so that we can see any valid message parsed correctly.